components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -163,75 +165,177 @@ export async function getIsSwapSupported (network: BraveWallet.NetworkInfo): Pro
 }
 
 export function refreshVisibleTokenInfo (currentNetwork: BraveWallet.NetworkInfo) {
-  return async (dispatch: Dispatch) => {
+  return async (dispatch: Dispatch, getState: () => State) => {
     const { braveWalletService } = getAPIProxy()
+    const { wallet: { networkList } } = getState()
+
+    const getVisibleAssets = await Promise.all(networkList.map(async (network) => {
+      // Creates a network's Native Asset if not returned
+      const nativeAsset: BraveWallet.BlockchainToken = {
+        contractAddress: '',
+        decimals: network.decimals,
+        isErc20: false,
+        isErc721: false,
+        logo: network.iconUrls[0] ?? '',
+        name: network.symbolName,
+        symbol: network.symbol,
+        // MULTICHAIN: Change visible back to false once getUserAssets returns
+        // SOL and FIL by default.
+        visible: network.coin === BraveWallet.CoinType.SOL || network.coin === BraveWallet.CoinType.FIL,
+        tokenId: '',
+        coingeckoId: '',
+        chainId: network.chainId
+      }
 
-    // Selected Network's Native Asset
-    const nativeAsset: BraveWallet.BlockchainToken = {
-      contractAddress: '',
-      decimals: currentNetwork.decimals,
-      isErc20: false,
-      isErc721: false,
-      logo: currentNetwork.iconUrls[0] ?? '',
-      name: currentNetwork.symbolName,
-      symbol: currentNetwork.symbol,
-      visible: false,
-      tokenId: '',
-      coingeckoId: ''
-    }
-
-    const visibleTokensInfo = await braveWalletService.getUserAssets(currentNetwork.chainId)
-    const visibleAssets: BraveWallet.BlockchainToken[] = visibleTokensInfo.tokens.length === 0 ? [nativeAsset] : visibleTokensInfo.tokens
+      // Get a list of user tokens for each coinType and network.
+      const getTokenList = network.chainId === BraveWallet.LOCALHOST_CHAIN_ID &&
+        network.coin === BraveWallet.CoinType.SOL || network.coin === BraveWallet.CoinType.FIL
+        // Since LOCALHOST's chainId is shared between coinType networks,
+        // this check will make sure we create the correct Native Asset for
+        // that network.
+        ? { tokens: [nativeAsset] } // MULTICHAIN: We do not yet support getting userAssets for FIL and SOL
```

Comment:
Is there an open issue we can reference here?

Reply:
Yes, referenced a link to the issue in the comment.